WordPress/Classes/Utility/ManagedObjectContextFactory.h

```diff
@@ -0,0 +1,18 @@
+#import <CoreData/CoreData.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/// A type that's used to handle how `NSManagedObjectContext` objects are created and how the data in them are saved.
+@protocol ManagedObjectContextFactory<NSObject>
```

Comment:
I'm pretty bad at naming things and am happy to rename these types from "factory" to other better names.

Reply:
IMHO `PersistentContainerProvider` might be a worthwhile substitute? Then you could have `NSPersistentContainerProvider` and `LegacyPersistentContainerProvider`? But I don't feel strongly about it 😅

Reply:
This protocol doesn't create a `NSPersistentContainer` instance though. The protocol takes a container instance and uses it to create `NSManagedObjectContext` instances, so maybe we shouldn't use "PersistentContainer" in its name? Also, we may not want to use the `NS` type prefix, which kind of indicates the type comes from the Foundation framework?

What do you think about, renaming the protocol to `ManagedObjectContextProviderProtocol`, and its two implementations to `LegacyManagedObjectContextProvider` and `ManagedObjectContextProvider`?